gemfirexd/core/src/main/java/com/pivotal/gemfirexd/internal/engine/sql/execute/SnappyPrepareResultSet.java

```diff
@@ -267,4 +272,27 @@ public FormatableBitSet getValidColumns() {
   @Override
   public void closeRowSource() {
   }
+
+  public ResultDescription makeResultDescription(String statementType) {
+    String[] colNames = firstResultHolder.getColumnNames();
+    int[] colTypes = firstResultHolder.getColumnTypes();
+    String[] tableNames = firstResultHolder.getTableNames();
+    DataTypeDescriptor[] dtds = firstResultHolder.getDtds();
+    if (colNames == null || colTypes == null || !(colNames.length > 0) || !(colTypes.length > 0)) {
+      throw new IllegalStateException("colnames and colTypes are required");
+    }
+
+    GenericResultDescription resultDescription = new GenericResultDescription(
+        new ResultColumnDescriptor[colTypes.length], statementType);
+
+    // TODO: KN remove hard coding
```

Comment:
prepare() method uses current schema, Can't that be used here?

Reply:
On the gemxd side when things start failing at the parser level itself then the metadata has to be fetched from the leadnode. The queries metadata cannot be found out from the current schema. So we are bringing the metadat from the lead node. Does this make thing clear or am I missing your point?